File: runtime/flutter/widgets/focusScope.ts
import { IList } from "../../dart/core/list";
import { IDiagnosticable } from "../foundation/diagnosticable";
import { DiagnosticLevel } from "../foundation/diagnosticLevel";
import { IDiagnosticPropertiesBuilder } from "../foundation/diagnosticPropertiesBuilder";
import { IDiagnosticsNode } from "../foundation/diagnosticsNode";
import { DiagnosticsTreeStyle } from "../foundation/diagnosticsTreeStyle";
import { IKey } from "../foundation/key";
import { IKeyEvent } from "../services/keyEvent";
import { IRawKeyEvent } from "../services/rawKeyEvent";
import { IBuildContext } from "./buildContext";
import { IFocus } from "./focus";
import { IFocusNode } from "./focusNode";
import { IFocusScopeNode } from "./focusScopeNode";
import { KeyEventResult } from "./keyEventResult";
import { IState } from "./state";
import { IStatefulElement } from "./statefulElement";
import { IWidget } from "./widget";
declare const flutter: {
    widgets: {
        focusScope: (
            this: void,
            focusScope: IFocusScope,
            props: {
                autofocus: boolean;
                canRequestFocus?: boolean | undefined;
                debugLabel?: string | undefined;
                key?: IKey | undefined;
                node?: IFocusScopeNode | undefined;
                onFocusChange?: (value: boolean) => void | undefined;
                onKey?: (
                    node: IFocusNode,
                    event: IRawKeyEvent
                ) => KeyEventResult | undefined;
                onKeyEvent?: (
                    node: IFocusNode,
                    event: IKeyEvent
                ) => KeyEventResult | undefined;
                skipTraversal?: boolean | undefined;
                child: IWidget;
            }
        ) => IFocusScope;
        focusScopeWithExternalFocusNode: (props: {
            autofocus: boolean;
            key?: IKey | undefined;
            onFocusChange?: (value: boolean) => void | undefined;
            child: IWidget;
            focusScopeNode: IFocusScopeNode;
        }) => IFocusScope;
        focusScopeOf: (context: IBuildContext) => IFocusScopeNode;
    };
};
export interface IFocusScope {
    child: IWidget;
    focusNode: IFocusNode | undefined;
    autofocus: boolean;
    onFocusChange: (value: boolean) => void | undefined;
    includeSemantics: boolean;
    key: IKey | undefined;
    createState: () => IState<IFocus>;
    debugFillProperties: (properties: IDiagnosticPropertiesBuilder) => void;
    getOnKeyEvent: () => (
        node: IFocusNode,
        event: IKeyEvent
    ) => KeyEventResult | undefined;
    getOnKey: () => (
        node: IFocusNode,
        event: IRawKeyEvent
    ) => KeyEventResult | undefined;
    getCanRequestFocus: () => boolean;
    getSkipTraversal: () => boolean;
    getDescendantsAreFocusable: () => boolean;
    getDebugLabel: () => string | undefined;
    createElement: () => IStatefulElement;
    toStringShort: () => string;
    getHashCode: () => number;
    toStringShallow: (props: {
        joiner: string;
        minLevel: DiagnosticLevel;
    }) => string;
    toStringDeep: (props: {
        minLevel: DiagnosticLevel;
        prefixLineOne: string;
        prefixOtherLines?: string | undefined;
    }) => string;
    toDiagnosticsNode: (props?: {
        name?: string | undefined;
        style?: DiagnosticsTreeStyle | undefined;
    }) => IDiagnosticsNode;
    debugDescribeChildren: () => IList<IDiagnosticsNode>;
    toString: (props: { minLevel: DiagnosticLevel }) => string;
}
export class FocusScope implements IFocus, IDiagnosticable {
    public readonly child: IWidget = undefined as any;
    public readonly focusNode: IFocusNode | undefined = undefined as any;
    public readonly autofocus: boolean = undefined as any;
    public readonly onFocusChange: (value: boolean) => void | undefined =
        undefined as any;
    public readonly includeSemantics: boolean = undefined as any;
    public readonly key: IKey | undefined = undefined as any;
    public constructor(props: {
        autofocus?: boolean;
        canRequestFocus?: boolean | undefined;
        debugLabel?: string | undefined;
        key?: IKey | undefined;
        node?: IFocusScopeNode | undefined;
        onFocusChange?: (value: boolean) => void | undefined;
        onKey?: (
            node: IFocusNode,
            event: IRawKeyEvent
        ) => KeyEventResult | undefined;
        onKeyEvent?: (
            node: IFocusNode,
            event: IKeyEvent
        ) => KeyEventResult | undefined;
        skipTraversal?: boolean | undefined;
        child: IWidget;
    }) {
        flutter.widgets.focusScope(this, {
            ...focusScopeDefaultProps,
            ...props,
        });
    }
    public static withExternalFocusNode(props: {
        autofocus: boolean;
        key?: IKey | undefined;
        onFocusChange?: (value: boolean) => void | undefined;
        child: IWidget;
        focusScopeNode: IFocusScopeNode;
    }): IFocusScope {
        return flutter.widgets.focusScopeWithExternalFocusNode(props);
    }
    public static of(context: IBuildContext): IFocusScopeNode {
        return flutter.widgets.focusScopeOf(context);
    }
    private readonly _dart_createState: () => IState<IFocus> = undefined as any;
    private readonly _dart_debugFillProperties: (
        properties: IDiagnosticPropertiesBuilder
    ) => void = undefined as any;
    private readonly _dart_getOnKeyEvent: () => (
        node: IFocusNode,
        event: IKeyEvent
    ) => KeyEventResult | undefined = undefined as any;
    private readonly _dart_getOnKey: () => (
        node: IFocusNode,
        event: IRawKeyEvent
    ) => KeyEventResult | undefined = undefined as any;
    private readonly _dart_getCanRequestFocus: () => boolean = undefined as any;
    private readonly _dart_getSkipTraversal: () => boolean = undefined as any;
    private readonly _dart_getDescendantsAreFocusable: () => boolean =
        undefined as any;
    private readonly _dart_getDebugLabel: () => string | undefined =
        undefined as any;
    private readonly _dart_createElement: () => IStatefulElement =
        undefined as any;
    private readonly _dart_toStringShort: () => string = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    private readonly _dart_toStringShallow: (props: {
        joiner: string;
        minLevel: DiagnosticLevel;
    }) => string = undefined as any;
    private readonly _dart_toStringDeep: (props: {
        minLevel: DiagnosticLevel;
        prefixLineOne: string;
        prefixOtherLines?: string | undefined;
    }) => string = undefined as any;
    private readonly _dart_toDiagnosticsNode: (props?: {
        name?: string | undefined;
        style?: DiagnosticsTreeStyle | undefined;
    }) => IDiagnosticsNode = undefined as any;
    private readonly _dart_debugDescribeChildren: () => IList<IDiagnosticsNode> =
        undefined as any;
    private readonly _dart_toString: (props: {
        minLevel: DiagnosticLevel;
    }) => string = undefined as any;
    public createState(): IState<IFocus> {
        return this._dart_createState();
    }
    public debugFillProperties(properties: IDiagnosticPropertiesBuilder): void {
        return this._dart_debugFillProperties(properties);
    }
    public getOnKeyEvent(): (
        node: IFocusNode,
        event: IKeyEvent
    ) => KeyEventResult | undefined {
        return this._dart_getOnKeyEvent();
    }
    public getOnKey(): (
        node: IFocusNode,
        event: IRawKeyEvent
    ) => KeyEventResult | undefined {
        return this._dart_getOnKey();
    }
    public getCanRequestFocus(): boolean {
        return this._dart_getCanRequestFocus();
    }
    public getSkipTraversal(): boolean {
        return this._dart_getSkipTraversal();
    }
    public getDescendantsAreFocusable(): boolean {
        return this._dart_getDescendantsAreFocusable();
    }
    public getDebugLabel(): string | undefined {
        return this._dart_getDebugLabel();
    }
    public createElement(): IStatefulElement {
        return this._dart_createElement();
    }
    public toStringShort(): string {
        return this._dart_toStringShort();
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
    public toStringShallow(props: {
        joiner?: string;
        minLevel?: DiagnosticLevel;
    }): string {
        return this._dart_toStringShallow({
            ...toStringShallowDefaultProps,
            ...props,
        });
    }
    public toStringDeep(props: {
        minLevel?: DiagnosticLevel;
        prefixLineOne?: string;
        prefixOtherLines?: string | undefined;
    }): string {
        return this._dart_toStringDeep({
            ...toStringDeepDefaultProps,
            ...props,
        });
    }
    public toDiagnosticsNode(props?: {
        name?: string | undefined;
        style?: DiagnosticsTreeStyle | undefined;
    }): IDiagnosticsNode {
        return this._dart_toDiagnosticsNode(props);
    }
    public debugDescribeChildren(): IList<IDiagnosticsNode> {
        return this._dart_debugDescribeChildren();
    }
    public toString(props: { minLevel?: DiagnosticLevel }): string {
        return this._dart_toString({
            ...toStringDefaultProps,
            ...props,
        });
    }
}
const focusScopeDefaultProps = {
    autofocus: false,
};
const toStringShallowDefaultProps = {
    joiner: ", ",
    minLevel: DiagnosticLevel.debug,
};
const toStringDeepDefaultProps = {
    minLevel: DiagnosticLevel.debug,
    prefixLineOne: "",
};
const toStringDefaultProps = {
    minLevel: DiagnosticLevel.info,
};

File: lib/cfr/builtins/libs/flutter/widgets/focusScope.dart
import 'dart:core' as _fac9;

import 'package:flutter/src/foundation/diagnostics.dart' as _eaf3;
import 'package:flutter/src/foundation/key.dart' as _ab4a;
import 'package:flutter/src/services/hardware_keyboard.dart' as _9e29;
import 'package:flutter/src/services/raw_keyboard.dart' as _94e8;
import 'package:flutter/src/widgets/focus_manager.dart' as _b485;
import 'package:flutter/src/widgets/focus_scope.dart' as _c024;
import 'package:flutter/src/widgets/framework.dart' as _e2dc;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedFocusScope extends _36c2.VMManagedBox<_c024.FocusScope> {VMManagedFocusScope({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) { table  [
'child'
] = _36c2.maybeBoxObject<_e2dc.Widget>(object: vmObject.child, hydroState: hydroState, table: _36c2.HydroTable());
table  [
'focusNode'
] = _36c2.maybeBoxObject<_b485.FocusNode?>(object: vmObject.focusNode, hydroState: hydroState, table: _36c2.HydroTable());
table  [
'autofocus'
] = vmObject.autofocus;

table  [
'includeSemantics'
] = vmObject.includeSemantics;
table  [
'key'
] = _36c2.maybeBoxObject<_ab4a.Key?>(object: vmObject.key, hydroState: hydroState, table: _36c2.HydroTable());
table  [
'createState'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_e2dc.State>(object: vmObject.createState(), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'debugFillProperties'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.debugFillProperties(_36c2.maybeUnBoxAndBuildArgument<_eaf3.DiagnosticPropertiesBuilder, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState));
return []; } );
table  [
'getOnKeyEvent'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { final returnValue = vmObject.onKeyEvent;if(returnValue!= null){return [returnValue,];}return []; } );
table  [
'getOnKey'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { final returnValue = vmObject.onKey;if(returnValue!= null){return [returnValue,];}return []; } );
table  [
'getCanRequestFocus'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.canRequestFocus,]; } );
table  [
'getSkipTraversal'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.skipTraversal,]; } );
table  [
'getDescendantsAreFocusable'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.descendantsAreFocusable,]; } );
table  [
'getDebugLabel'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { final returnValue = vmObject.debugLabel;if(returnValue!= null){return [returnValue,];}return []; } );
table  [
'createElement'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_e2dc.StatefulElement>(object: vmObject.createElement(), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'toStringShort'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toStringShort(),]; } );
table  [
'getHashCode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.hashCode,]; } );
table  [
'toStringShallow'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return [vmObject.toStringShallow(joiner: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'joiner'
] : null, minLevel: _36c2.maybeUnBoxEnum(values: _eaf3.DiagnosticLevel.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'minLevel'
] : null)),]; } );
table  [
'toStringDeep'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 

return [vmObject.toStringDeep(minLevel: _36c2.maybeUnBoxEnum(values: _eaf3.DiagnosticLevel.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'minLevel'
] : null), prefixLineOne: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'prefixLineOne'
] : null, prefixOtherLines: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'prefixOtherLines'
] : null),]; } );
table  [
'toDiagnosticsNode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return [_36c2.maybeBoxObject<_eaf3.DiagnosticsNode>(object: vmObject.toDiagnosticsNode(name: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'name'
] : null, style: _36c2.maybeUnBoxEnum(values: _eaf3.DiagnosticsTreeStyle.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'style'
] : null)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'toString'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toString(minLevel: _36c2.maybeUnBoxEnum(values: _eaf3.DiagnosticLevel.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'minLevel'
] : null)),]; } ); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _c024.FocusScope vmObject;

 }

class RTManagedFocusScope extends _c024.FocusScope implements _36c2.Box<_c024.FocusScope> {RTManagedFocusScope({required _fac9.bool autofocus, _fac9.bool? canRequestFocus, _fac9.String? debugLabel, _ab4a.Key? key, _b485.FocusScopeNode? node, void Function(_fac9.bool value)? onFocusChange, _b485.KeyEventResult Function(_b485.FocusNode node, _94e8.RawKeyEvent event)? onKey, _b485.KeyEventResult Function(_b485.FocusNode node, _9e29.KeyEvent event)? onKeyEvent, _fac9.bool? skipTraversal, required _e2dc.Widget child, required this.table, required this.hydroState}) : super(autofocus: autofocus,canRequestFocus: canRequestFocus,debugLabel: debugLabel,key: key,node: node,onFocusChange: onFocusChange,onKey: onKey,onKeyEvent: onKeyEvent,skipTraversal: skipTraversal,child: child) { table  [
'vmObject'
] = vmObject;
table  [
'unwrap'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [unwrap()]; } );
table  [
'child'
] = _36c2.maybeBoxObject(object: this.child, hydroState: hydroState, table: _36c2.HydroTable());
table  [
'focusNode'
] = _36c2.maybeBoxObject(object: focusNode, hydroState: hydroState, table: _36c2.HydroTable());
table  [
'autofocus'
] = _36c2.maybeBoxObject(object: this.autofocus, hydroState: hydroState, table: _36c2.HydroTable());

table  [
'includeSemantics'
] = _36c2.maybeBoxObject(object: includeSemantics, hydroState: hydroState, table: _36c2.HydroTable());
table  [
'key'
] = _36c2.maybeBoxObject(object: this.key, hydroState: hydroState, table: _36c2.HydroTable());
table['_dart_createState'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [
    _36c2.maybeBoxObject(
        object: super.createState(),
        hydroState: hydroState,
        table: _36c2.HydroTable())
  ];
});
table['_dart_debugFillProperties'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  super.debugFillProperties(_36c2.maybeUnBoxAndBuildArgument<
      _eaf3.DiagnosticPropertiesBuilder,
      _fac9.dynamic>(luaCallerArguments[1], parentState: hydroState));
  return [];
});
table['_dart_getCanRequestFocus'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [super.canRequestFocus];
});
table['_dart_getSkipTraversal'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [super.skipTraversal];
});
table['_dart_getDescendantsAreFocusable'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [super.descendantsAreFocusable];
});
table['_dart_getDebugLabel'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [super.debugLabel];
});
table['_dart_createElement'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [
    _36c2.maybeBoxObject(
        object: super.createElement(),
        hydroState: hydroState,
        table: _36c2.HydroTable())
  ];
});
table['_dart_toStringShort'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [
    _36c2.maybeBoxObject(
        object: super.toStringShort(),
        hydroState: hydroState,
        table: _36c2.HydroTable())
  ];
});
table['_dart_getHashCode'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [super.hashCode];
});
table['_dart_toStringShallow'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [
    _36c2.maybeBoxObject(
        object: super.toStringShallow(
            joiner:
                _36c2.maybeUnBoxAndBuildArgument<_fac9.String, _fac9.dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['joiner']
                        : null,
                    parentState: hydroState),
            minLevel: _36c2.maybeUnBoxEnum(
                values: _eaf3.DiagnosticLevel.values,
                boxedEnum: luaCallerArguments.length >= 2
                    ? luaCallerArguments[1]['minLevel']
                    : null)),
        hydroState: hydroState,
        table: _36c2.HydroTable())
  ];
});
table['_dart_toStringDeep'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [
    _36c2.maybeBoxObject(
        object: super.toStringDeep(
            minLevel: _36c2.maybeUnBoxEnum(
                values: _eaf3.DiagnosticLevel.values,
                boxedEnum: luaCallerArguments.length >= 2
                    ? luaCallerArguments[1]['minLevel']
                    : null),
            prefixLineOne:
                _36c2.maybeUnBoxAndBuildArgument<_fac9.String, _fac9.dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['prefixLineOne']
                        : null,
                    parentState: hydroState),
            prefixOtherLines:
                _36c2.maybeUnBoxAndBuildArgument<_fac9.String?, _fac9.dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['prefixOtherLines']
                        : null,
                    parentState: hydroState)),
        hydroState: hydroState,
        table: _36c2.HydroTable())
  ];
});
table['_dart_toDiagnosticsNode'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [
    _36c2.maybeBoxObject(
        object: super.toDiagnosticsNode(
            name:
                _36c2.maybeUnBoxAndBuildArgument<_fac9.String?, _fac9.dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['name']
                        : null,
                    parentState: hydroState),
            style: _36c2.maybeUnBoxEnum(
                values: _eaf3.DiagnosticsTreeStyle.values,
                boxedEnum: luaCallerArguments.length >= 2
                    ? luaCallerArguments[1]['style']
                    : null)),
        hydroState: hydroState,
        table: _36c2.HydroTable())
  ];
});
table['_dart_debugDescribeChildren'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [
    _36c2.maybeBoxObject(
        object: super.debugDescribeChildren(),
        hydroState: hydroState,
        table: _36c2.HydroTable())
  ];
});
table['_dart_toString'] =
    _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
  return [
    _36c2.maybeBoxObject(
        object: super.toString(
            minLevel: _36c2.maybeUnBoxEnum(
                values: _eaf3.DiagnosticLevel.values,
                boxedEnum: luaCallerArguments.length >= 2
                    ? luaCallerArguments[1]['minLevel']
                    : null)),
        hydroState: hydroState,
        table: _36c2.HydroTable())
  ];
}); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

_c024.FocusScope unwrap() => this;
_c024.FocusScope get vmObject => this;
@_fac9.override _e2dc.State<_c024.Focus> createState() { 
_36c2.Closure closure = table["createState"];
return _36c2.maybeUnBoxAndBuildArgument<_e2dc.State<_c024.Focus>, _c024.Focus>(closure.dispatch([table],parentState: hydroState)[0], parentState: hydroState); } 
@_fac9.override void debugFillProperties(properties) { super.debugFillProperties(properties);
_36c2.Closure closure = table["debugFillProperties"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
@_fac9.override _fac9.bool get canRequestFocus { 
_36c2.Closure closure = table["getCanRequestFocus"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
@_fac9.override _fac9.bool get skipTraversal { 
_36c2.Closure closure = table["getSkipTraversal"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
@_fac9.override _fac9.bool get descendantsAreFocusable { 
_36c2.Closure closure = table["getDescendantsAreFocusable"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
@_fac9.override _fac9.String? get debugLabel { 
_36c2.Closure closure = table["getDebugLabel"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
@_fac9.override _e2dc.StatefulElement createElement() { 
_36c2.Closure closure = table["createElement"];
return _36c2.maybeUnBoxAndBuildArgument<_e2dc.StatefulElement, _fac9.dynamic>(closure.dispatch([table],parentState: hydroState)[0], parentState: hydroState); } 
@_fac9.override _fac9.String toStringShort() { 
_36c2.Closure closure = table["toStringShort"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
@_fac9.override _fac9.String toStringShallow({_fac9.String joiner = ', ', _eaf3.DiagnosticLevel minLevel = _eaf3.DiagnosticLevel.debug}) { 
_36c2.Closure closure = table["toStringShallow"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
@_fac9.override _fac9.String toStringDeep({_eaf3.DiagnosticLevel minLevel = _eaf3.DiagnosticLevel.debug, _fac9.String prefixLineOne = '', _fac9.String? prefixOtherLines}) { 
_36c2.Closure closure = table["toStringDeep"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
@_fac9.override _eaf3.DiagnosticsNode toDiagnosticsNode({_fac9.String? name, _eaf3.DiagnosticsTreeStyle? style}) { 
_36c2.Closure closure = table["toDiagnosticsNode"];
return _36c2.maybeUnBoxAndBuildArgument<_eaf3.DiagnosticsNode, _fac9.dynamic>(closure.dispatch([table],parentState: hydroState)[0], parentState: hydroState); } 
@_fac9.override _fac9.List<_eaf3.DiagnosticsNode> debugDescribeChildren() { 
_36c2.Closure closure = table["debugDescribeChildren"];
return _36c2.maybeUnBoxAndBuildArgument<_fac9.List<_eaf3.DiagnosticsNode>, _eaf3.DiagnosticsNode>(closure.dispatch([table],parentState: hydroState)[0], parentState: hydroState); } 
@_fac9.override _fac9.String toString({_eaf3.DiagnosticLevel minLevel = _eaf3.DiagnosticLevel.info}) { 
_36c2.Closure closure = table["__tostring"];
return closure.dispatch([table],parentState: hydroState)[0]; } 
 }
void loadFocusScope({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { table  [
'focusScope'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 




_36c2.Closure? unpackedonFocusChange=luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'onFocusChange'
] : null;
_36c2.Closure? unpackedonKey=luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'onKey'
] : null;
_36c2.Closure? unpackedonKeyEvent=luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'onKeyEvent'
] : null;


return  [RTManagedFocusScope(table: luaCallerArguments  [
0
], hydroState: hydroState, autofocus: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'autofocus'
] : null, canRequestFocus: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'canRequestFocus'
] : null, debugLabel: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'debugLabel'
] : null, key: _36c2.maybeUnBoxAndBuildArgument<_ab4a.Key?, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'key'
] : null, parentState: hydroState), node: _36c2.maybeUnBoxAndBuildArgument<_b485.FocusScopeNode?, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'node'
] : null, parentState: hydroState), onFocusChange: unpackedonFocusChange != null ? (value) => unpackedonFocusChange.dispatch([luaCallerArguments[0],value],parentState:hydroState,) : null , onKey: unpackedonKey != null ? (node, event) => _36c2.maybeUnBoxEnum(values: _b485.KeyEventResult.values, boxedEnum: unpackedonKey.dispatch([luaCallerArguments[0],node,event],parentState:hydroState,)[0]) : null , onKeyEvent: unpackedonKeyEvent != null ? (node, event) => _36c2.maybeUnBoxEnum(values: _b485.KeyEventResult.values, boxedEnum: unpackedonKeyEvent.dispatch([luaCallerArguments[0],node,event],parentState:hydroState,)[0]) : null , skipTraversal: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'skipTraversal'
] : null, child: _36c2.maybeUnBoxAndBuildArgument<_e2dc.Widget, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'child'
] : null, parentState: hydroState))]; } );
table  [
'focusScopeWithExternalFocusNode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 

_36c2.Closure? unpackedonFocusChange=luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'onFocusChange'
] : null;

return [_36c2.maybeBoxObject<_c024.FocusScope>(object: _c024.FocusScope.withExternalFocusNode(autofocus: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'autofocus'
] : null, key: _36c2.maybeUnBoxAndBuildArgument<_ab4a.Key?, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'key'
] : null, parentState: hydroState), onFocusChange: unpackedonFocusChange != null ? (value) => unpackedonFocusChange.dispatch([luaCallerArguments[0],value],parentState:hydroState,) : null , child: _36c2.maybeUnBoxAndBuildArgument<_e2dc.Widget, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'child'
] : null, parentState: hydroState), focusScopeNode: _36c2.maybeUnBoxAndBuildArgument<_b485.FocusScopeNode, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'focusScopeNode'
] : null, parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'focusScopeOf'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_b485.FocusScopeNode>(object: _c024.FocusScope.of(_36c2.maybeUnBoxAndBuildArgument<_e2dc.BuildContext, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
_36c2.registerBoxer<_c024.FocusScope>(boxer: ({required _c024.FocusScope vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedFocusScope(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 
